test/correctness/specialize.cpp

```diff
@@ -443,6 +442,58 @@ int main(int argc, char **argv) {
         }
     }
 
+    {
+        // Check specialization on a more complex expression used in a select.
+        ImageParam im(Int(32), 2);
+        Param<int> p;
+        Expr test = (p > 73) || (p*p + p + 1 == 0);
+
+        Func f;
+        Var x;
+        f(x) = select(test, im(x, 0), im(0, x));
+        f.specialize(test);
+
+        Buffer<int> input1(10, 1);
+        Buffer<int> input2(1, 10);
+        im.set(input1);
+        p.set(100);
+        f.realize(10);
+
+        im.set(input2);
+        p.set(-100);
+        f.realize(10);
```

Comment:
Add a comment on what the test is supposed to do (e.g. throw an error if the condition doesn't simplify, etc.)